micro_dl/utils/masks.py

```diff
@@ -94,3 +95,65 @@ def create_unimodal_mask(input_image, str_elem_size=3):
     thr_image = binary_opening(input_image > thr, str_elem)
     mask = binary_erosion(thr_image, str_elem)
     return mask
+
+
+def get_unet_border_weight_map(annotation, w0=10, sigma=5):
+    """
+    Generate the weight map for borders as specified in the UNet paper for an annotation.
+    Parameters
+    ----------
+    annotation: array-like
+        A 2D array of shape (image_height, image_width) contains annotation with each class labeled as an integer.
+
+    Returns
+    -------
+    array-like
+        A 2D array of shape (image_height, image_width)
+
+    TODO: Calculate boundaries directly and calculate distance from boundary of cells to another
+    TODO: The below method only works for UNet Segmentation only
+    """
+
+    assert annotation.dtype == np.uint8, "datatype expected uint8, it is {}".format(annotation.dtype)
+    # class balance weights w_c(x)
+    unique_values = np.unique(annotation).tolist()
+    weight_map = [0] * len(unique_values)
+    for index, unique_value in enumerate(unique_values):
+        mask = np.zeros((annotation.shape[0], annotation.shape[1]), dtype=np.float64)
+        mask[mask == unique_value] = 1
+        if mask.sum() != 0:
+            weight_map[index] = 1 / mask.sum()
+        else:
+            weight_map[index] = 1
+
+    # this normalization is important - foreground pixels must have weight 1
+    weight_map = [i / max(weight_map) for i in weight_map]
+
+    wc = np.zeros((mask.shape[0], mask.shape[1]), dtype=np.float64)
+    for index, unique_value in enumerate(unique_values):
+        wc[mask == unique_value] = weight_map[index]
+
+    # cells instances for distance computation
+    labeled_array, _ = scipy.ndimage.measurements.label(annotation)
+    # cells distance map
+    border_loss_map = np.zeros((annotation.shape[0], annotation.shape[1]), dtype=np.float64)
+    distance_maps = np.zeros((annotation.shape[0], annotation.shape[1], np.max(labeled_array)), dtype=np.float64)
+    if np.max(labeled_array) >= 2:
+        for index, label in enumerate(range(1, np.max(labeled_array) + 1)):
```

Comment:
just a thought, simplify may be, not combine enumerate and range, for index in range(np.max(labeled_array): label = index + 1